lib/adf_diag.py

```diff
@@ -1032,32 +1114,45 @@ def derive_variables(self, res=None, vars_to_derive=None, ts_dir=None, overwrite
 
         """
 
+        #Loop through derived variables
         for var in vars_to_derive:
             print(f"\t - deriving time series for {var}")
 
-            #Check whether there are parts to derive from and if there is an associated equation
-            vres = res.get(var, {})
-            if "derivable_from" in vres:
-                constit_list = vres['derivable_from']
-            else:
-                print("WARNING: No constituents listed in defaults config file, moving on")
-                continue
+            #Grab list of constituents for this variable
+            constit_list = constit_dict[var]
 
-            #Grab all required time series files for derived var
+            #Grab all required time series files for derived variable
             constit_files = []
             for constit in constit_list:
-                if glob.glob(os.path.join(ts_dir, f"*.{constit}.*.nc")):
-                    constit_files.append(glob.glob(os.path.join(ts_dir, f"*.{constit}.*"))[0])
+                #Check if the constituent file is present, if so add it to list
+                if glob.glob(os.path.join(ts_dir, f"*{hist_str}*.{constit}.*.nc")):
+                    constit_files.append(glob.glob(os.path.join(ts_dir, f"*{hist_str}*.{constit}.*"))[0])
```

Comment:
I might have a check here for when `hist_str = None`:

```suggestion
                if hist_str:
                    const_glob_str = f"*{hist_str}*.{constit}.*.nc"
                else:
                    const_glob_str = f"*.{constit}.*.nc"
                #end if
                if glob.glob(os.path.join(ts_dir, const_glob_str)):
                    constit_files.append(glob.glob(os.path.join(ts_dir, const_glob_str ))[0])
```